libyul/backends/wasm/polyfill/Keccak.yul

```diff
@@ -18,7 +18,332 @@
 
 // NOTE: This file is used to generate `ewasmPolyfills/Keccak.h`.
 
+// reset the context
+function keccak_reset(context_offset:i32) {
+	// state context
+	//	typedef struct {
+	//		union {                                 // state:
+	//			uint8_t b[200];                     // 8-bit bytes
+	//			uint64_t q[25];                     // 64-bit words
+	//		} st;
+	//		int pt, rsiz, mdlen;                    // these don't overflow
+	//	} sha3_ctx_t;
+	memset(i32.add(context_offset, 0:i32), 0:i32, 200:i32)
+	//	c->mdlen = mdlen;
+	//	c->rsiz = 200 - 2 * mdlen;
+	//	c->pt = 0;
+	i64.store(mdlen_offset(context_offset), 32)
+	i64.store(rsiz_offset(context_offset), 136) // 200 - 2 * mdlen = 136
+	i64.store(pt_offset(context_offset), 0)
+}
+function mdlen_offset(context_offset:i32) -> offset:i32 {
+	offset :=  i32.add(context_offset, 200:i32)
+}
+function rsiz_offset(context_offset:i32) -> offset:i32 {
+	offset :=  i32.add(context_offset, 208:i32)
+}
+function pt_offset(context_offset:i32) -> offset:i32 {
+	offset :=  i32.add(context_offset, 216:i32)
+}
+function rounds_consts_offset(context_offset:i32) -> offset:i32 {
+	offset :=  i32.add(context_offset, 224:i32)
+}
+function keccakf_rotc_offset(context_offset:i32) -> offset:i32 {
+	offset :=  i32.add(context_offset, 416:i32)
+}
+function keccakf_piln_offset(context_offset:i32) -> offset:i32 {
+	offset :=  i32.add(context_offset, 440:i32)
+}
+function bc_offset(context_offset:i32) -> offset:i32 {
+	offset :=  i32.add(context_offset, 464:i32)
+}
```

Comment:
These should have docstrings saying what they refer to. Names like `mdlen`, `rsiz` or `pt` are pretty cryptic.

Even in tiny_sha they are not really explained. Knowing only https://keccak.team/keccak_specs_summary.html only some of them look familiar at this point.

Reply:
Also:
```suggestion
function mdlen_offset(context_offset:i32) -> offset:i32 {
	offset := i32.add(context_offset, 200:i32)
}
function rsiz_offset(context_offset:i32) -> offset:i32 {
	offset := i32.add(context_offset, 208:i32)
}
function pt_offset(context_offset:i32) -> offset:i32 {
	offset := i32.add(context_offset, 216:i32)
}
function rounds_consts_offset(context_offset:i32) -> offset:i32 {
	offset := i32.add(context_offset, 224:i32)
}
function keccakf_rotc_offset(context_offset:i32) -> offset:i32 {
	offset := i32.add(context_offset, 416:i32)
}
function keccakf_piln_offset(context_offset:i32) -> offset:i32 {
	offset := i32.add(context_offset, 440:i32)
}
function bc_offset(context_offset:i32) -> offset:i32 {
	offset := i32.add(context_offset, 464:i32)
}
```